File: 2.programming_technology/C_Programming/Practice/basic/q1_calculator.c
//Implement a choice based arithmetic calculator (1-add,2-sub,3-mul,4-div etc)
#include <stdio.h>
int main()
{
    int num1,num2,ch;
    while(1){

        printf("\nEnter First Number : ");
        scanf("%d",&num1);
        printf("\nEnter Second Number : ");
        scanf("%d",&num2);

        printf("1. Add \n");
        printf("2. Sub \n");
        printf("3. Mul \n");
        printf("4. Div \n");
        printf("5. Exit \n");
        printf("Please choose any operation from above choices : ");
        scanf("%d",&ch);

        switch(ch){
            case 1 : printf("Addition of Two Numbers (%d + %d) is : %d ",num1,num2,num1+num2);
                     break;

            case 2 : printf("Subtraction of Two Numbers (%d - %d) is : %d ",num1,num2,num1-num2);
                     break;

            case 3 : printf("Multiplication of Two Numbers (%d * %d) is : %d ",num1,num2,num1*num2);
                     break;

            case 4 : float div= (float)num1 / num2 ; 
                    printf("Division of Two Numbers (%d / %d) is : %.2f ",num1,num2,(float)num1 / num2);
                     break;

            case 5 : break;

            default : printf("Please Choose appropriate options \n");
                     break;

        }

    }
    


    return 0;
}

File: 2.programming_technology/Python_Programming/Practice/oops/examples/class_obj/q1_calculator.py
#Q1 Python Program to Create a Class which Performs Basic Calculator Operations

class Calculator:

    def __init__ (self,num1,num2):
        self.num1 = num1
        self.num2 = num2

    def sum(self):
        res= self.num1 + self.num2
        print("Addition is : ",res)

    def sub(self):
        res = self.num1 - self.num2   
        print("Subtraction is  : ",res)

    def mul(self):
        res = self.num1 * self.num2
        print("Multiplication is  :",res)

    def div(self):    
        res = self.num1 / self.num2
        print("Division is :",res)

def menu():
    print("[1]. Addition ")
    print("[2]. Subtraction ")
    print("[3]. Multiplication ")
    print("[4]. Division ")

   
number1 = int(input("Enter Number 1 : "))
number2 = int(input("Enter Number 2 : "))

cal = Calculator(number1,number2)

while True:
    menu()
    ch = int(input("Choose operation to perform : "))
    if ch == 1:
        cal.sum()    
    elif ch == 2:
        cal.sub()
    elif ch == 3:
        cal.mul()
    elif ch == 4:
        cal.div()
    else:
        print("Choose appropriate option")    


    